commercetools-models/src/main/java/io/sphere/sdk/products/commands/ProductImageUploadCommand.java

```diff
@@ -0,0 +1,104 @@
+package io.sphere.sdk.products.commands;
+
+import com.fasterxml.jackson.databind.JavaType;
+import io.sphere.sdk.client.HttpRequestIntent;
+import io.sphere.sdk.commands.CommandImpl;
+import io.sphere.sdk.http.HttpMethod;
+import io.sphere.sdk.http.UrlQueryBuilder;
+import io.sphere.sdk.json.SphereJsonUtils;
+import io.sphere.sdk.products.ByIdVariantIdentifier;
+import io.sphere.sdk.products.Product;
+
+import javax.annotation.Nullable;
+import java.io.File;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+public final class ProductImageUploadCommand extends CommandImpl<Product> {
+    private final String productId;
+    @Nullable
+    private final Integer variant;
+    @Nullable
+    private final String sku;
+    @Nullable
+    private final String filename;
+    @Nullable
+    private final Boolean staged;
+    private final File body;
+    @Nullable
+    private String givenContentType;
```

Comment:
I would suggest to add `final` here and then add this property to the constructor. This is more consistent with our existing code.